js/templates/comment-form.handlebars

```diff
@@ -94,98 +94,111 @@
     ">
 
-    {{!-- ================ SOCIAL PICTURE && TEXT AREA ================ --}}
-    <div style="display: table; width: 100%">
-      {{#unless is_anon}}
+  {{!-- ================ SOCIAL PICTURE && TEXT AREA ================ --}}
+  <div style="display: table; width: 100%">
+    {{#unless is_anon}}
       <span style="display: table-cell">
         {{view profilePicView}}
       </span>
-      {{/unless}}
-      <span style="display: table-cell; width: 100%">
+    {{/unless}}
+    <span style="display: table-cell; width: 100%">
         <textarea
           class="form-control"
           id="comment_form_textarea"
           rows="1"
           style="
-          display: inline-block;
+            display: inline-block;
             width: 100%;
             font-size: 16px; {{!-- 16px to prevent zooming on iOS. http://stackoverflow.com/questions/2989263/disable-auto-zoom-in-input-text-tag-safari-on-iphone --}}
             line-height: 23px;
             vertical-align: middle;
-          "
+            "
           name="txt"
           type="text"
           maxlength="400"
           {{!-- placeholder="Submit your own opinions here. Shorter is better - break up your ideas. You are not responding to anyone else directly - comments are sent out randomly to all participants."  --}}
           placeholder="{{s.writePrompt}}"
           {{#if shouldAutofocusOnTextarea}}autofocus{{/if}}
 
-          ></textarea>
-        {{!-- <p
-          id="commentSentAlert"
-          class="protip"
-          style="display: none;"
-          role="alert">
-            <strong>
-              <i
-                class="fa fa-bullhorn">
-              </i>
-              {{s.commentSent}}
-              </strong>
-        </p> --}}
+        ></textarea>
+      {{!-- <p
+        id="commentSentAlert"
+        class="protip"
+        style="display: none;"
+        role="alert">
+          <strong>
+            <i
+              class="fa fa-bullhorn">
+            </i>
+            {{s.commentSent}}
+            </strong>
+      </p> --}}
       </span>
-    </div>
+  </div>
 
-    {{!-- ================ CONNECT FACEBOOK / TWITTER ================ --}}
-    <div
-      id="socialButtonsCommentForm"
-      style="
+  {{!-- ================ CONNECT FACEBOOK / TWITTER ================ --}}
+  <div
+    id="socialButtonsCommentForm"
+    style="
         display: none;
         margin-top: 10px;
         ">
-      <div class="protip"><p><i class="fa fa-microphone"></i>&nbsp; {{s.connectToPostPrompt}}</p></div>
-      {{#unless hasFacebook}}
-        <button id="facebookButtonCommentForm" class="facebookButton" style="padding-top: 0px;">
-          <i class="svgIcon" style="
+    <div class="protip"><p><i class="fa fa-microphone"></i>&nbsp; {{s.connectToPostPrompt}}</p></div>
+    {{#unless hasEMail}}
+      <button id="emailButtonCommentForm" class="emailButton" style="padding-top: 0px;">
```

Comment:
@urakagi Can you please clarify what the email button is for?

Reply:
E-mail button is for email login. Our system cannot be fully connected to social networks, so we need an SNS-independent login method. Thus I make email login and authorization works and puts a button on the `clientParticipant`.